src/runtime/plugins/i18n.ts

```diff
@@ -224,8 +224,12 @@ export default defineNuxtPlugin(async nuxt => {
           _getLocaleCookie(nuxt.ssrContext, { ...nuxtI18nOptions.detectBrowserLanguage, localeCodes })
         composer.setLocaleCookie = (locale: string) =>
           _setLocaleCookie(locale, nuxt.ssrContext, nuxtI18nOptions.detectBrowserLanguage || undefined)
-        composer.onBeforeLanguageSwitch = nuxtI18nOptions.onBeforeLanguageSwitch
-        composer.onLanguageSwitched = nuxtI18nOptions.onLanguageSwitched
+
+        // @ts-ignore
+        composer.onBeforeLanguageSwitch = (o, n, i, ctx) => nuxt.callHook('i18n:before-lang-switch', o, n, i, ctx)
```

Comment:
if it's possible, I hope you would resolve type. 🙏

Reply:
I do want to propose changing the parameters for these functions to be an object with the values in properties instead, so it would be

```ts
type Props<Context = unknown> = {
  oldLocale: string,
  newLocale: string,
  initialSetup: boolean,
  context: Context
}

export type BeforeLanguageSwitchHandler = <C = unknown>(prop: Props<C>) => string | void

// then..
composer.onBeforeLanguageSwitch = (p) => nuxt.callHook('i18n:before-lang-switch', p)
```

The function type should not be an issue, however passing `'i18n:before-lang-switch'` becomes a warning since the built-in hooks doesn't have that option and I have seen other modules use ts-ignore for it. Perhaps I need to dig deeper into docs and see what can be done!

Reply:
Thanks! I've understood your proposal.

>I do want to propose changing the parameters for these functions to be an object with the values in properties instead

Indeed, It makes sense to pass objects instead of params. With objects, it is less likely to break the function's argument signature if we add something we want to pass to function.

we want to rename the nuxt hooks defined for the new i18n module as follows:

- `'i18n:before-lang-switch'` -> `'i18n:beforeLocaleSwitch'`
- `'i18n:locale-switched'` -> `'i18n:localeSwitched'`

If there are naming guidelines for nuxt hooks, they should be followed.

The following are lifecycle hooks:
https://nuxt.com/docs/api/advanced/hooks#lifecycle-hooks

Reply:
In that case, should I rename the existing `i18n:extend-messages` hook to be camel case as well?

Reply:
Yeah!
The topic differs from this PR, so we should handle it with another PR.